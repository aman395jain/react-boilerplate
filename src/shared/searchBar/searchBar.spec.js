import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import SearchBox from "./searchBar";

configure({ adapter: new Adapter() });
let props, events;
beforeEach(() => {
  props = {
    event: {
      keyCode: 27
    }
  };
});
describe("SearchBox", () => {
  const component = shallow(<SearchBox debug />);

  it('should render correctly in "debug" mode', () => {
    expect(component).toMatchSnapshot();
    expect(component.find(".input-without-text")).toBeDefined();
    expect(component.find("input")).toBeDefined();
  });

  it("should test the handle escape key to be true", () => {
    // component.find("input").simulate("onKeyDown", props.event);
    component.instance().handleKeyEsc(props.event);
    expect(component.state().escapeFlag).toBeTruthy;
  });

  it("should test the handle escape key to be false", () => {
    component.instance().handleKeyEsc({ keyCode: "01" });
    expect(component.state().escapeFlag).toBeFalsy;
  });

  it("should test the search With Debounce Time", () => {
    component.instance().searchWithDebounceTime(
      (events = {
        target: {
          value: "test"
        }
      })
    );
    expect(component.state().text).toBe("test");
  });
});
