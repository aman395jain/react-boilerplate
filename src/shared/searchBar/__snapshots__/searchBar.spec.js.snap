// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBox
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="search-input-without-text"
        >
          <input
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Search text"
            type="text"
            value=""
          />
        </div>,
        <ul
          className="searchBar-dropdown"
        />,
      ],
      "className": "seachBar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Search text"
            type="text"
            value=""
          />,
          "className": "search-input-without-text",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "onKeyDown": [Function],
            "placeholder": "Search text",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "searchBar-dropdown",
        },
        "ref": null,
        "rendered": Array [],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="search-input-without-text"
          >
            <input
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Search text"
              type="text"
              value=""
            />
          </div>,
          <ul
            className="searchBar-dropdown"
          />,
        ],
        "className": "seachBar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Search text"
              type="text"
              value=""
            />,
            "className": "search-input-without-text",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "onKeyDown": [Function],
              "placeholder": "Search text",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "searchBar-dropdown",
          },
          "ref": null,
          "rendered": Array [],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
